test/lspconfig_spec.lua

```diff
@@ -319,7 +319,7 @@ describe('lspconfig', function()
         local _ = lspconfig.sumneko_lua
         local _ = lspconfig.tsserver
         lspconfig.rust_analyzer.setup {}
-        return lspconfig.available_servers()
+        return require("lspconfig.util").available_servers()
```

Comment:
This seems like a breaking change to me. `lspconfig.available_servers()` should still exist

Reply:
Well at the moment, the `lspconfig.lua` simply sets up the metatable for `lspconfig.configs` and nothing else. If we add this, then it will setup a metatable + a single function that can actually be in utils.

Reply:
> Then the logic is simple. The portion of plugin simply sets up LSPs, with some commands in `plugin/` and some useful functions in `/utils`.

Yeah I agree it makes sense to remove it, I'm just thinking about not breaking parity with `master`. This will for sure break some user configs and plugins out there, although easily remedied.

> Also, too many PRs are being merged that potentially clash with this. Requires a bit of reshuffling

Yeah there seems to be a lot happening with this big PR still being open, I'll eject myself from further comments, let's get this merged haha

Reply:
> Yeah I agree it makes sense to remove it, I'm just thinking about not breaking parity with `master`. This will for sure break some user configs and plugins out there, although easily remedied.
> 
Well this is changing/simplifying the core mechanism of the project. We could merge this as a breaking change, with a tag for previous versions?